working/macros/feature-specification.md

```diff
@@ -279,6 +279,118 @@ is not user visible. For example, if two macros are applied to two methods in
 the same class, there is no way for those macros to interfere with each other
 such that the application order can be detected.
 
+### Augmentation library structure and ordering
+
+It is important that expanding macros in a given library always results in a
+consistent augmentation library. In particular, multiple tools should be able to
+run the same macros with the same inputs, and get the same augmentation library.
+This allows debugging and stack traces to work consistently, and be meaningful
+and useful.
+
+It is also important that if a declaration is added in Phase 1, the source
+offsets to that declaration should not change after Phase 2 or 3 run. This means
+that tools don't need to update source offsets after each phase.
+
+Another important consideration is that in Phase 2, the ordering of macros is
+user perceptable, and so augmentation results need to be serialized to disk -
```

Comment:
I wouldn't say that the results *need* to be serialized. An implementation should be free to not do that. But I would say that it must be *possible* to serialize them.